### PR TITLE
refactor(ivy): cleanup the public API for core/testing

### PR DESCRIPTION
**Pull request by vicb**
/cc @IgorMinar @mhevery 

http://test/OCL:208737684:BASE:208763730:1534305206082:abdfffb

### CODE REVIEW

File: packages/core/testing/src/test_bed_common.ts [hunk 1]
```diff
@@ -38,3 +42,95 @@ export type TestModuleMetadata = {
   schemas?: Array<SchemaMetadata|any[]>,
   aotSummaries?: () => any[],
 };
+
+/**
+ * Static methods implemented by the `TestBedViewEngine` and `TestBedRender3`
+ */
+export interface TestBedConstructor {
```
Comment (mhevery) [inline packages/core/testing/src/test_bed_common.ts]: `FooConstructor` is the convention from TypeScript for this. However in our case you can't do `new TestBedConstructor` so the above does not make sense. What do you think about renaming it to `TestBedStatic`? I think that makes more sense.

PS: I know I was the one who suggested that name but now I am not so sure.

**Review by mhevery — APPROVED** [review]

File: packages/core/testing/src/test_bed_common.ts (hunk 1 above)
Comment (vicb) [reply packages/core/testing/src/test_bed_common.ts]: I actually was thinking about `TestBedStatic` but picked `TestBedConstructor` to please you :)
I'll update that once test are passing.

**Review by IgorMinar — APPROVED** [review]
this looks much better. please make the ci lint pass. thanks

**Comment by vicb** [comment]
@IgorMinar Agreed. Looks much nicer with an API break !

**Pull request closed without merging**